api/magic_links/routes.py

```diff
@@ -0,0 +1,159 @@
+        """
+        Creates a json dict of minimal link claims and generates a
+        signed token of those claims and adds the token as value
+        on the returned dict
+        :param account: An Account object holding the user's details
+        :param redirect_url: String url to redirect to
+        :return: A dictionary containing the link claims and signed token
+        """
+        new_link_json = {
+            "accountId": account.id,
+            "iat": int(datetime.now().timestamp()),
+            "exp": int(
+                (
+                    datetime.now()
+                    + timedelta(
+                        days=env.config.get("MAGIC_LINK_EXPIRY_DAYS", 0),
+                        minutes=1,
+                    )
+                ).timestamp()
+            ),
+            "redirectUrl": redirect_url,
+        }
+        new_link_json.update({"token": create_token(new_link_json)})
+        return new_link_json
+
+    def _set_unique_keyed_record(self, value) -> str | None:
+        """
+        Attempts to save the link claims with a unique short key
+        that doesn't already exist (and returns None if it fails)
+        :param value: A json dictionary of link attributes
+        :return: The unique short key for the link hash (or None)
+        """
+        key = self._make_short_key()
+        created = self.redis_mlinks.setex(
+            key, env.config.get("MAGIC_LINK_EXPIRY_SECONDS", 0), value
+        )
+        if created == 1:
+            return key
+
+    def search(self):
+        """
+        GET /magic-links endpoint
+        :return: Json Response
+        """
+        return Response(json.dumps(self._links), mimetype="application/json")
+
+    def use(self, link_id: str):
+        """
+        GET /magic-links/{link_id} endpoint
+        If the link_id matches a valid link key, this then:
+        - creates a session,
+        - sets a session_id cookie in the client
+        - sets a session_token cookie in the client
+        - deletes the link record from redis
+        - then finally, redirects to the redirect_url
+        If no matching valid link is found returns a 404 error message
+        :param link_id: String short key for the link
+        :return: 302 Redirect / 404 Error
+        """
+        link_hash = self.redis_mlinks.get(link_id)
+        if link_hash:
+            link = json.loads(link_hash)
+            self.redis_mlinks.delete(link_id)
+
+            # Check link is not expired
+            if link.get("exp") > int(datetime.now().timestamp()):
+                return AuthSessionView.create_session_and_redirect(
+                    account_id=link.get("accountId"),
+                    redirect_url=link.get("redirectUrl"),
+                )
+            return error_response(404, "Link expired")
+        return error_response(404, "Link expired or invalid")
+
+    def create(self, email: str, redirect_url: str = "https://google.com"):
```

Comment:
could we get rid of this default, or use something a bit more generic like https://example.com ?